src/cls/rgw/cls_rgw.cc

```diff
@@ -1132,6 +1246,10 @@ int rgw_bucket_complete_op(cls_method_context_t hctx, bufferlist *in, bufferlist
 		       __func__, escape_str(idx).c_str(), rc);
         return rc;
       }
+      bufferlist empty;
+      rc = record_duplicate_entry(hctx, idx, entry.key, &empty, header.resharding_in_logrecord());
+      if (rc < 0)
+        return rc;
```

Comment:
can replace this and the `cls_cxx_map_remove_key()` call above with `remove_entry()`